megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -618,6 +621,99 @@ protected void calculateFiringTurn() {
                     logger.debug(shooter.getDisplayName() + " - Detailed Best Firing Plan: " +
                             plan.getDebugDescription(true));
 
+                    // If the shooter is a ground unit (including VTOLs and infantry) and only
+                    // shooting at a single Mech target with all weapons, consider making an
+                    // aimed shot if the target is shut down or the attacker has a targeting
+                    // computer. Alternatively, consider using the called shots optional rule to
+                    // adjust the hit table to something more favorable.
+
+                    boolean isShutdownShot = false;
+                    boolean isAimedShot = false;
+                    boolean isCalledShot = false;
+                    int advancedTargetingThreshold = ToHitData.IMPOSSIBLE;
+                    int maxAdvancedTargetNumber = 10;
+                    int aimLocation = Mech.LOC_NONE;
+                    int locationDestruction = 999;
+                    int calledShotDirection = CalledShot.CALLED_NONE;
+
+                    int targetID = plan.get(0).getTarget().getId();
```

Comment:
Any time you call "plan.get(0).whatever" without checking the get(0) for null, you're risking a null pointer exception. Let's make sure to check for that prior to executing any of the added code since I see a lot of calls to it in there.